data/adapters/VARIANT_EFFECTS_variant_gene_adapter.py

```diff
@@ -0,0 +1,142 @@
+
+
+class VARIANTEFFECTSAdapter:
+    ALLOWED_LABELS = ['variant', 'variant_gene']
+    SOURCE = 'IGVF'
+    CHUNK_SIZE = 6500
+
+    def __init__(self, filepath, label, source_url, writer: Optional[Writer] = None, **kwargs):
+        if label not in self.ALLOWED_LABELS:
+            raise ValueError(
+                f'Invalid label. Allowed values: {", ".join(self.ALLOWED_LABELS)}')
+        self.filepath = filepath
+        self.label = label
+        self.source_url = source_url
+        self.file_accession = source_url.split('/')[-2]
+        self.writer = writer
+        self.type = 'node' if label == 'variant' else 'edge'
+        self.gene_validator = GeneValidator()
+
+        fileset = FileFileSet(self.file_accession,
+                              writer=None, label='igvf_file_fileset')
+        props, _, _ = fileset.query_fileset_files_props_igvf(
+            self.file_accession, replace=False)
+        self.simple_sample_summaries = props['simple_sample_summaries']
+        self.biosample_term = props['samples']
+        self.treatments_term_ids = props['treatments_term_ids']
+        self.method = props['method']
+
+    def process_file(self):
+        self.writer.open()
+
+        with open(self.filepath, 'r') as f:
+            reader = csv.reader(f, delimiter='\t')
+            next(reader)
+            chunk = []
+            for i, row in enumerate(reader, 1):
+                chunk.append(row)
+                if i % VARIANTEFFECTSAdapter.CHUNK_SIZE == 0:
+                    self.process_chunk(chunk)
+                    chunk = []
+
+            if chunk != []:
+                self.process_chunk(chunk)
+
+        self.writer.close()
+
+    def process_chunk(self, chunk):
+        spdi_to_variant = {}
+        spdi_to_row = {}
+        skipped_spdis = []
+        for row in chunk:
+            gene = row[7]
+            if not self.gene_validator.validate(gene):
+                raise ValueError(
+                    f'{gene} is not a valid gene.')
+
+            spdi = row[0]
+            variant, skipped_message = load_variant(spdi)
+
+            if variant:
+                normalized_spdi = variant['spdi']
+                spdi_to_variant[normalized_spdi] = variant
+                if normalized_spdi not in spdi_to_row:
+                    spdi_to_row[normalized_spdi] = []
+                spdi_to_row[normalized_spdi].append(row)
+
+            if skipped_message is not None:
+                skipped_spdis.append(skipped_message)
+
+        if skipped_spdis:
+            print(f'Skipped {len(skipped_spdis)} variants:')
+            for skipped in skipped_spdis:
+                print(f"  - {skipped['variant_id']}: {skipped['reason']}")
+            with open('./skipped_variants.jsonl', 'a') as out:
+                for skipped in skipped_spdis:
+                    out.write(json.dumps(skipped) + '\n')
+
+        loaded_variants = bulk_check_spdis_in_arangodb(spdi_to_variant.keys())
```

Comment:
I would probably move this forward, before load_variant function, because you don't need to validate + normalize variant if it's already loaded. It's depending on the data also - if the majority is new variants then it's fine to leave it here

Reply:
But I need to get the normalized SPDI to check the database with?